pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/ExistingRecordPatch.java

```diff
@@ -208,6 +209,14 @@ Object handleSingleNameValue() {
         if (Patch.patch(patch.getId(), patch.getUser(), patch.getNameValue(), dbElement)) {
             nameValues.addAll(setWorkflowRelatedFields(patch));
             exportToESWithId(patch.getId(), patch.getNameValue());
+            if (patch.getNameValue().getName().equals("oD.tissueReceived")) {
+                if (StringUtils.isBlank(patch.getParentId()) && StringUtils.isNotBlank(patch.getDdpParticipantId())) {
+                    patch.setParentId(patch.getDdpParticipantId());
+                }
+                if (StringUtils.isNotBlank(patch.getParentId())) {
+                    triggerParticipantEvent(ddpInstance, patch, new Value(null, EventTypeDao.EVENT, "TISSUE_RECEIVED"));
+                }
```

Comment:
Would recommend making the literal strings here static variables.
Also, there are some checks here for things being blank that affect logic. Some logging might be helpful if we are trying to figure out what the specific case that we are dealing with is or if there is an unexpected error condition.